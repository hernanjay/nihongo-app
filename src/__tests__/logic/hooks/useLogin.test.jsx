import { expect, it } from "vitest";

const login = async (email, password) => {
  const response = await fetch(
    `https://aws-nihongo-api.onrender.com/api/users/login`,
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        email,
        password,
      }),
    }
  );

  const json = await response.json();

  if (!response.ok) {
    return {
      status: response.status,
      error: json.error,
    };
  }

  if (response.ok) {
    return json;
  }
};

describe("Test login function", () => {
  it("should first call the login function", async () => {
    const user = await login("admin", "admin");
    expect(user).not.toBeNull();
  });
  it("User should successfuly login", async () => {
    const user = await login("nan@awsys-i.com", "Wew@123321");
    expect(user).toBe(user);
  });
  it("User should unsuccessfuly login due to wrong email", async () => {
    const user = await login("admin", "admin");
    console.log(`Test 3 value is : ${user.status} , ${user.error}`);
    expect(user.status).toBe(400);
    expect(user.error).toBe("Incorrect email!");
  });
});
