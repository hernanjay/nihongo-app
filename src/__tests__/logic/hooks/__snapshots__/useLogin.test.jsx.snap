// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Login Page > should render Login component correctly 1`] = `
<React.Fragment>
  <Accordion>
    <AccordionItem>
      <h2>
        <AccordionButton
          data-testid="AccordionButtontestid"
        >
          <Box
            as="span"
            flex="1"
            textAlign="left"
          >
            Section 1 title
          </Box>
          <AccordionIcon />
        </AccordionButton>
      </h2>
      <AccordionPanel
        pb={4}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </AccordionPanel>
    </AccordionItem>
    <AccordionItem>
      <h2>
        <AccordionButton>
          <Box
            as="span"
            flex="1"
            textAlign="left"
          >
            Section 2 title
          </Box>
          <AccordionIcon />
        </AccordionButton>
      </h2>
      <AccordionPanel
        pb={4}
      >
        Hidden Content
      </AccordionPanel>
    </AccordionItem>
  </Accordion>
</React.Fragment>
`;
